pkg/jwt/public_keys.go

```diff
@@ -75,23 +75,27 @@ func (j *JWKSGetter) fetch() {
 	}
 }
 
-type singleKeyStore struct {
-	keyID string
-	key   interface{}
+// derivedPublicKeyStore provides public key by looking up into underlying private key store.
+type derivedPublicKeyStore struct {
+	keyID           string
+	privateKeyStore PrivateKeyGetter
 }
 
-func (s *singleKeyStore) Key(id string) interface{} {
-	if id == s.keyID {
-		return s.key
+// NewDerivedPublicKeyStore returns a PublicKeyGetter that allows obtaining a single, derived from private key with a defined id.
+func NewDerivedPublicKeyStore(privateKeyStore PrivateKeyGetter, keyID string) PublicKeyGetter {
+	return &derivedPublicKeyStore{
+		keyID:           keyID,
+		privateKeyStore: privateKeyStore,
 	}
-	log.Error("not found")
-	return nil
 }
 
-// NewSingleKeyStore returns a KeyGetter that allows obtaining a single key with a defined id.
-func NewSingleKeyStore(key interface{}, keyID string) KeyGetter {
-	return &singleKeyStore{
-		keyID: keyID,
-		key:   key,
+// Key returns public key with matching id.
+func (d *derivedPublicKeyStore) Key(id string) interface{} {
+	if id == d.keyID {
+		privateKey := d.privateKeyStore.Key(d.keyID)
+		if privateKey != nil {
+			return privateKey.Public()
+		}
 	}
+	return nil
```

Comment:
nit (for less identation):
```suggestion
        if id != d.keyID {
                return nil
        }
        privateKey := d.privateKeyStore.Key(d.keyID)
        if privateKey != nil {
                return privateKey.Public()
        }
        return nil
```